src/pages/HomePage.tsx

```diff
@@ -11,43 +11,52 @@ interface ICryptoCoin {
   current_price: string
 }
 export default function HomePage() {
-  const [data, setData] = useState<ICryptoCoin[]>([])
   const URL: string = `https://api.coingecko.com/api/v3/coins/markets?vs_currency=eur&order=market_cap_desc&per_page=100&page=1&sparkline=false`
 
-  useEffect(() => {
-    fetch(URL)
-      .then(date => date.json())
-      .then(setData)
-      .catch(err => {
-        console.error(err)
-      })
-  }, [URL])
+  const fetchCoins = async () => {
+    const response = await fetch(URL)
+    return response.json()
+  }
+  const { data, isLoading, isError, error, isIdle } = useQuery<
+    ICryptoCoin[],
+    Error
+  >('coins', fetchCoins)
 
-  const topFiveCoins = data.slice(-5)
+  const topFiveCoins = data?.slice(-5)
 
   return (
     <Box m="30px">
-      <Card>
-        <Box p="30px">
-          <Carousel>
-            <Grid container columnGap="15px" rowGap="15px">
-              {topFiveCoins.map((crypto: ICryptoCoin) => (
-                <Grid item key={crypto.id}>
-                  <CoinCard
-                    border=" border border-purple"
-                    title={crypto.name}
-                    icon={crypto.image}
-                    price={crypto.current_price}
-                  />
+      {isIdle ? (
+        'Not ready...'
```

Comment:
Not sure if this is best practice, looks a bit too much. I think checking `isLoading` for displaying a loading state and otherwise displaying the data is enough.